### PR TITLE
iptables: for listener using filter chain

### PR DESCRIPTION
**Pull request by silentdai**
Context
Inbound outbound listener split using filter chain instead of 2nd level listeners has extra requirement on iptables

Goal
The goal is the same as before: 
When client envoy send request to server envoy(through pod ip:port) and
** client envoy happens to be the same as server envoy**,
the request should be captured.

Previously it is captured by the unique capture port 15001, with the inout listener split it should be captured by 15006, 
which is the new inbound virtual listener port.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by silentdai** [comment]
notes: @costinm @fpesce @howardjohn This is not supposed to address the problem a moment ago :(

File: tools/packaging/common/istio-iptables.sh [hunk 1]
```diff
@@ -410,7 +410,7 @@ fi
 if [ -z "${DISABLE_REDIRECTION_ON_LOCAL_LOOPBACK-}" ]; then
   # Redirect app calls back to itself via Envoy when using the service VIP or endpoint
   # address, e.g. appN => Envoy (client) => Envoy (server) => appN.
-  iptables -t nat -A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -j ISTIO_REDIRECT
```
Comment (costinm) [inline tools/packaging/common/istio-iptables.sh]: Not sure if this is correct or fixes the problem. More comments in the PR description would help. Did you manually test that the loop is gone ? With or without split chains ? 

The IN filter has a 'original dst', so envoy will still connect to itself, and hit this rule in a loop.

File: tools/packaging/common/istio-iptables.sh [hunk 2]
```diff
@@ -410,7 +410,7 @@ fi
...
+  iptables -t nat -A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -j ISTIO_IN_REDIRECT
```
Comment (silentdai) [inline tools/packaging/common/istio-iptables.sh]: @rshriram This line is the core change in this PR :)
This rule is supposed to  the case that client envoy send to server envoy. 
istio should capture this packet by redirect the packet to inbound listener(ISTIO_IN_REDIRECT stands for istio_*inbound*_redirect. 
If the inbound and outbound listener were not split, or inbound listener and outbound listener are equivalent(in/out listener are almost the same before introducing filterchains in virtual in/out bound listeners #15713), this bug is not hurting.

File: tools/packaging/common/istio-iptables.sh (hunk 1 above)
Comment (silentdai) [reply tools/packaging/common/istio-iptables.sh]: Sorry this PR is **NOT** the approach to fix the infinite loop we chatted yesterday. As stated in my comment...

This PR is suppose to 
- support the combination of split and inboundfilterchain to fix the bug only when filterchain is on
- not break anything when listener split is off (ISTIO_REREDIRECT and ISTIO_IN_REDIRECT is the same)

**Comment by costinm** [comment]
My concern is that the fix for the loop is effectively removing this line.

Also this would only apply if split chains are enabled - did we remove the
option to not use split chains ?

Let's hold on this until the loop is fixed - that's the release blocking /
critical bug - and it appears the fix will
be on this line actually.

BTW - we need to test if just setting the DISALBE_REDIRECTION doesn't fix
the problem in the first place...

**Comment by silentdai** [comment]
gentle ping
This is a simple change. The core one line change is self explaining.

**Comment by silentdai** [comment]
#15788 
#15790 
#15791

File: tools/packaging/common/istio-iptables.sh (hunk 2 above)
Comment (costinm) [reply tools/packaging/common/istio-iptables.sh]: Got it.

**Comment by silentdai** [comment]
> that's the release blocking /critical bug - and it appears the fix will be on this line actually.

Actually that's why I extract the iptables change from #15713 
I'd like to provide the picture before a fix from @fpesce 
We need to understand what iptables should be, and fix the infinite loop above it. Otherwise @fpesce will be fixing a bug on top of an wrong foundation.

**Comment by silentdai** [comment]
@costinm  Thanks!

**Comment by silentdai** [comment]
This is required by combined filter chain. It might fix a bug that envoy upstream traffic  is intercepted by outbound listener. Although it is not reported yet.

**Pull request merged**